packages/core/src/messaging/eventBus.ts

```diff
@@ -38,7 +36,7 @@ export class EventBus {
         this.#eventEmitter.removeListener(eventName, callback, once);
     }
 
-    dispatch(eventName: EventName, data?: unknown) {
+    dispatch(eventName: EventName, data: unknown) {
```

Comment:
It's possible to dispatch an event without a payload, data should be optionnal. I've done it in a few codebases.